### PR TITLE
[16.0][IMP] stock_account: refactoring for checking owner_id

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
This PR adds a refactoring of the owner_id check within the stock_account module.
The purpose of this refactoring is to prepare for a custom module that will value inventory based on specific ownership. This refactoring allows for extending to the functions as needed.

@qrtl

### CODE REVIEW

File: addons/stock_account/models/stock_move_line.py [hunk 1]
```diff
@@ -72,3 +72,26 @@ def _create_correction_svl(self, move, diff):
...
+
+    @api.model
+    def _should_exclude_for_valuation(self):
+        """
+        Determines if this move line should be excluded from valuation based on its ownership.
+        :return: True if the move line's owner is different from the company's partner (indicating
+                it should be excluded from valuation), False otherwise.
+        """
+        self.ensure_one()
+        if self.owner_id and self.owner_id != self.company_id.partner_id:
```
Comment (pfertyk) [inline addons/stock_account/models/stock_move_line.py:84]: In general, this structure:

```
if condition:
    return True
return False
```

can be replaced with:

```
return condition
```

That goes for the other 2 methods as well ;)

Comment (AungKoKoLin1997) [reply addons/stock_account/models/stock_move_line.py:84]: @pfertyk Actually, I considered your suggestion in the first place. But for design consistency, I decided to use currency approach. This is how odoo standard use for this kind of function.
https://github.com/odoo/odoo/blob/ed890131f5bff8264e152c77ae5d3757f2dd7bfb/addons/stock_account/models/stock_location.py#L25-L32

Comment (clesgow) [reply addons/stock_account/models/stock_move_line.py:84]: @AungKoKoLin1997 There are far more occurrences of what @pfertyk suggested in the code. Let's avoid bloating the code for nothing. :slightly_smiling_face:

File: addons/stock_account/models/stock_move_line.py [hunk 2]
```diff
@@ -72,3 +72,26 @@ def _create_correction_svl(self, move, diff):
...
+        return False
+    
+    @api.model
+    def _should_exclude_from_missing_qty_calculation(self):
+        """
+        Determines if this move line should be excluded from the calculation of the missing
+        quantity for valuation purposes based on its ownership.
+        """
+        self.ensure_one()
+        if not self.owner_id or self.owner_id == self.company_id.partner_id:
```
Comment (pfertyk) [inline addons/stock_account/models/stock_move_line.py:95]: BTW @AungKoKoLin1997 , this condition is the same as the one in the other method, just negated. Are you extracting 2 methods here on purpose, to be able to override each of them separately? If not, may I suggest using a single method instead? :)

Comment (AungKoKoLin1997) [reply addons/stock_account/models/stock_move_line.py:95]: @pfertyk I'm not quite sure I understand your point. Are you referring to` _should_exclude_for_valuation()` as the other method? I believe they do not share the same condition.

Comment (AungKoKoLin1997) [reply addons/stock_account/models/stock_move_line.py]: @pfertyk You're right. Thanks for suggestion. I updated the code.

File: addons/stock_account/models/product.py [hunk 3]
```diff
@@ -705,7 +705,7 @@ def _compute_average_price(self, qty_invoiced, qty_to_invoice, stock_moves, is_r
         # price to estimate the anglo saxon price unit.
         missing = qty_to_invoice - qty_valued
         for sml in stock_moves.move_line_ids:
-            if not sml.owner_id or sml.owner_id == sml.company_id.partner_id:
+            if sml._should_exclude_for_valuation():
```
Comment (pfertyk) [inline addons/stock_account/models/product.py:708]: Thank you for reducing both extracted methods to a single one :+1: 

In this place, however, we should negate the condition ;)
```suggestion
            if not sml._should_exclude_for_valuation():
```

**Comment by AungKoKoLin1997** [comment]
@clesgow I updated the commit message.

**Review by clesgow — APPROVED** [review]
Thank you!
@robodoo r+

**Pull request closed without merging**